packages/common/src/hardhat/plugins/ExtendedWeb3.js

```diff
@@ -0,0 +1,84 @@
+// Pluging notes:
+// The plugin adds 4 methods to the hre: getContract, findEvent, assertEventEmitted, assertEventNotEmitted.
+// It requires the web3 plugin to work. All contract instantiations are web3, not truffle.
+//
+// getContract usage:
+//   const { getContract } = hre;
+//   const Voting = getContract("Voting");
+//
+//   // Get deployed voting contract for network.
+//   const voting = await Voting.deployed();
+//
+//   // Create new Voting contract.
+//   const voting = await Voting.new(arg1, arg2, ...).send({ from: accounts[0] });
+//
+//   // Instantiate voting contract at address.
+//   const voting = Voting.at(someAddress);
+//
+// findEvent, assertEventEmitted, and assertEventNotEmitted usage:
+//   const { findEvent, assertEventEmitted, assertEventNotEmitted } = hre;
+//   ...
+//   const result = voting.methods.someMethod(arg1, arg2, ...).send({ from: accounts[0] });
+//   const matchOrUndefined = findEvent(result, voting, "SomeEventName",
+//                            eventReturnValues => eventReturnValues.someValueICareAbout === 5);
+//   await assertEventEmitted(result, voting, "SomeEventName",
+//                            eventReturnValues => eventReturnValues.someValueICareAbout === 5);
+//   await assertEventNotEmitted(result, voting, "SomeEventName",
+//                            eventReturnValues => eventReturnValues.someValueICareAbout === 5);
+//
+
+const { extendEnvironment } = require("hardhat/config");
+
+extendEnvironment((hre) => {
+  hre._artifactCache = {};
+  hre.getContract = (name) => {
+    if (!hre._artifactCache[name]) hre._artifactCache[name] = hre.artifacts.readArtifactSync(name);
+    const artifact = hre._artifactCache[name];
+
+    const deployed = async () => {
+      const deployment = await hre.deployments.get(name);
```

Comment:
This should read from both `networks/X.json` and the `deployments` folder right?

Reply:
It should do whatever hardhat deploy is configured to do!